service_api.md

```diff
@@ -0,0 +1,284 @@
+
+    class ProjectUpdateRequest {
+        String projectId
+        ProjectUpdateRequestBody requestBody
+        String requestId
+    }
+
+    class ProjectUpdateResponse {
+        String projectId
+        ProjectUpdateResponseBody responseBody
+        String requestId
+    }
+
+    class ProjectUpdateRequestBody {
+    }
+    <<interface>> ProjectUpdateRequestBody
+
+    class ProjectUpdateResponseBody {
+    }
+    <<interface>> ProjectUpdateResponseBody
+
+    class FundingInformation
+    FundingInformation ..|> ProjectUpdateRequestBody
+    FundingInformation ..|> ProjectUpdateResponseBody
+    ProjectUpdateRequest ..> ProjectUpdateRequestBody
+    ProjectUpdateResponse ..> ProjectUpdateResponseBody
+
+
+
+```
+
+### Update project contacts
+
+```mermaid
+---
+title: Update Project Contacts Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ProjectUpdateResponse~ update(ProjectUpdateRequest request)
+    }
+    <<interface>> AsyncProjectService
+    AsyncProjectService ..> ProjectUpdateRequest
+    AsyncProjectService ..> ProjectUpdateResponse
+
+    class ProjectUpdateRequest {
+        String projectId
+        ProjectUpdateRequestBody requestBody
+        String requestId
+    }
+
+    class ProjectUpdateResponse {
+        String projectId
+        ProjectUpdateResponseBody responseBody
+        String requestId
+    }
+
+    class ProjectUpdateRequestBody {
+    }
+    <<interface>> ProjectUpdateRequestBody
+
+    class ProjectUpdateResponseBody {
+    }
+    <<interface>> ProjectUpdateResponseBody
+
+    class ProjectContacts
+    ProjectContacts ..|> ProjectUpdateRequestBody
+    ProjectContacts ..|> ProjectUpdateResponseBody
+    ProjectUpdateRequest ..> ProjectUpdateRequestBody
+    ProjectUpdateResponse ..> ProjectUpdateResponseBody
+```
+
+### Update of a sub-resource with multiplicity
+
+Sometimes a resource contains more than one sub-resource of the same type. This happens, when there
+is a one-to-many relationship between the resource and its sub-resource. For example an experiment can
+contain one or more experimental groups.
+
+To enable clients to update a certain sub-resource, a combination of the parent resource's ID and the local 
+ID of the sub-resource is required.
+
+For example to update an `Experimental Group` of an `Experiment`, the API shall provide a method like `update(ExperimentalGroupUpdateRequest request)`. The object request needs to provide
+the required IDs to ensure that the service can perform the task.
+
+## Delete Example
+
+Deletion of a resource or sub-resource shall be possible with the resource ID or the combination of the resource ID
+and its sub-resource local ID.
+
+For example an `Experimental Group 1` in an `Experiment X` for `Project A` should be deletable by the client by e.g. calling `delete(ExperimentalGroupDeletionRequest request)`. The combination of project ID, experiment ID and experimental group ID ensures the
+identity of the resource manipulation,
```

Comment:
Same question as for the update of resources. Do I require all deletions to fail if one fails? How to recover from this?

Reply:
Please see my previous answer :)

Reply:
```suggestion
identity of the resource manipulation. Each request is transactional for the resource.
```